tests/test_simulation.py

```diff
@@ -665,6 +665,96 @@ def test_cached_image_accessors():
+    - overwrite=None: Renames the existing file and saves the new one.
+    """
+    sim1 = Simulation(seed=1)
+    sim2 = Simulation(seed=2)
+    sim3 = Simulation(seed=3)
+
+    # Create a tmp dir for this test output
+    with tempfile.TemporaryDirectory() as tmpdir_name:
+        # tmp filename
+        starfile = os.path.join(tmpdir_name, "og.star")
+        base, ext = os.path.splitext(starfile)
+
+        sim1.save(starfile, overwrite=True)
+
+        # Case 1: overwrite=True (should overwrite the existing file)
+        sim2.save(starfile, overwrite=True)
+
+        # Load and check if sim2 has overwritten sim1
+        sim2_loaded = RelionSource(starfile)
+        np.testing.assert_allclose(
+            sim2.images[:].asnumpy(),
+            sim2_loaded.images[:].asnumpy(),
+            atol=utest_tolerance(sim2.dtype),
+        )
+
+        # Check that metadata is unchanged.
+        check_metadata(sim2, sim2_loaded)
+
+        # Case 2: overwrite=False (should raise an overwrite error)
+        with raises(
+            ValueError,
+            match="File '.*' already exists; set overwrite=True to overwrite it",
+        ):
+            sim2.save(starfile, overwrite=False)
+
+        # case 3: overwrite=None (should rename the existing file and save im3 with original filename)
+        with caplog.at_level(logging.INFO):
+            sim3.save(starfile, overwrite=None)
+
+            # Check that the existing file was renamed and logged
+            assert f"Renaming {starfile}" in caplog.text
+
+            # Find the renamed file by checking the directory contents
+            renamed_file = None
+            for filename in os.listdir(tmpdir_name):
+                if filename.startswith("og_") and filename.endswith(".star"):
+                    renamed_file = os.path.join(tmpdir_name, filename)
+                    break
+
+            assert renamed_file is not None, "Renamed file not found"
+
+        # Load and check that sim3 was saved to the original path
+        sim3_loaded = RelionSource(starfile)
+        np.testing.assert_allclose(
+            sim3.images[:].asnumpy(),
+            sim3_loaded.images[:].asnumpy(),
+            atol=utest_tolerance(sim3.dtype),
+        )
+        check_metadata(sim3, sim3_loaded)
+
+        # Also check that the renamed file still contains sim2's data
+        sim2_loaded_renamed = RelionSource(renamed_file)
+        np.testing.assert_allclose(
+            sim2.images[:].asnumpy(),
+            sim2_loaded_renamed.images[:].asnumpy(),
+            atol=utest_tolerance(sim2.dtype),
+        )
+        check_metadata(sim2, sim2_loaded_renamed)
+
+
+def check_metadata(sim_src, relion_src):
+    """
+    Helper function to test if metadata fields in a Simulation match
+    those in a RelionSource.
+    """
+    for k, v in sim_src._metadata.items():
+        try:
+            np.testing.assert_array_equal(v, relion_src._metadata[k])
+        except AssertionError:
```

Comment:
You mentioned this in the office and reading it now, I'm just curious, which fields are causing trouble here? Maybe it is another oversight/issue we need to address in our sources. Thanks

Reply:
The `try: np.testing.assert_array_equal` checks the metadata fields that are strings or ints in both sources (.ie '_rlnSymmetryGroup', '_rlnClassNumber').  The `except` branch catches metadata fields that are all close (.ie all the float fields). The one field that requires casting away from a string is '_rlnAmplitude'. The amplitudes are loaded as string values in the `RelionSource`.

Reply:
Ahh, okay,  `_rlnAmplitude` is something we may need to fix on in another PR a little then.  (maybe not, but sounds like it).

If `_rlnAmplitude` was fixed do you think these could all be served by a single allclose with no try/except?

Reply:
The '_rlnSymmetryGroup' (strings) field would still give us trouble when trying to use allclose. I could use some other logic besides the try/except if you think that would be better. I could check the dtype or just use `assert_array_equals` for the specific symmetry group field...

Reply:
I see the string type promotion error now. Yeah, you are right.  Its not for this PR anyway. I was just curious in case it comes up again. Thanks!